com.avaloq.tools.ddk.xtext.builder/src/com/avaloq/tools/ddk/xtext/builder/MonitoredClusteringBuilderState.java

```diff
@@ -686,8 +686,8 @@ protected void storeBinaryResource(final Resource resource, final BuildData buil
     if (isBinaryModelStorageAvailable && resource instanceof StorageAwareResource && ((StorageAwareResource) resource).getResourceStorageFacade() != null
         && fileSystemAccess instanceof IFileSystemAccessExtension3) {
 
-      if (resource.getResourceSet() != null) {
-        CompletableFuture.runAsync(() -> {
+      CompletableFuture.runAsync(() -> {
+        if (resource.getResourceSet() != null) {

```

Comment:
You need to extract this lambda to an own method for Checkstyle to accept the change, otherwise, it looks good.